Runtime 5.0.1 release notes + Astronomer providers refactor (#683)
astronomer/docs: astro/runtime-release-notes.md

For instructions on how to upgrade, read [Upgrade Astro Runtime](upgrade-runtime.md). For general product release notes, go to [Astro Release Notes](release-notes.md). If you have any questions or a bug to report, reach out to [Astronomer Support](https://support.astronomer.io).

## Astro Runtime 5.0.1

- Rlease date: May 9, 2022
- Airflow version: 2.3.0

### Astronomer Providers 1.2.0

> Astro Runtime 5.0.1 includes v1.2.0 of the `astronomer-providers` package ([CHANGELOG](https://github.com/astronomer/astronomer-providers/blob/main/CHANGELOG.rst#110-2022-03-23)). This release includes 5 new [deferrable operators](deferrable-operators.md):

Author: ```suggestion
Astro Runtime 5.0.1 includes v1.2.0 of the `astronomer-providers` package ([CHANGELOG](https://astronomer-providers.readthedocs.io/en/stable/)). This release includes 5 new [deferrable operators](deferrable-operators.md):
```

    - `DataprocSubmitJobOperatorAsync`
    - `EmrContainerSensorAsync`
    - `EmrStepSensorAsync`
    - `EmrJobFlowSensorAsync`
    - `LivyOperatorAsync`

To access the source code of this package, visit the [Astronomer Providers GitHub repository](https://github.com/astronomer/astronomer-providers).

### Additional Improvements

> - Bumped the [`openlineage-airflow` dependency](https://openlineage.io/integration/apache-airflow/) to `v0.8.1`

Reviewer: ```suggestion
- Performance improvements that address various issues related to upgrading to Astro Runtime 5.0.0
- Bumped the [`openlineage-airflow` dependency](https://openlineage.io/integration/apache-airflow/) to v0.8.1
```

Adding a line-item for migration fixes! Was main reason we cut 5.0.1

> - Bumped the [`openlineage-airflow` provider package](https://openlineage.io/integration/apache-airflow/) to `v0.8.1`

Reviewer 2: ```suggestion
- Bumped the [`openlineage-airflow` dependency](https://openlineage.io/integration/apache-airflow/) to `v0.8.1`
```
